fastlane/lib/fastlane/actions/setup_ci.rb

```diff
@@ -7,19 +7,22 @@ def self.run(params)
           return
         end
 
-        case params[:provider]
-        when 'travis'
-          setup_keychain
+        case detect_provider(params)
         when 'circleci'
-          setup_keychain
           setup_output_paths
         end
+
+        setup_keychain
       end
 
       def self.should_run?(params)
         Helper.ci? || params[:force]
       end
 
+      def self.detect_provider(params)
```

Comment:
I feel this is beyond the scope of the PR and I feel it's weird that circleci receive a special treatment there. Either we can flirt the CI and then we should do it or we should keep the current behavior.

Reply:
Since the gist of this PR is that running `setup_ci` always installs the keychain, passing a `provider` param should be optional. That said, it might appear like _CircleCI_ receives special treatment but that is just because it is the only CI provider for which other setup step need to be performed. I didn't see the point in detecting every CI provider but then not acting on the result.